dlt/common/data_writers/escape.py

```diff
@@ -68,3 +68,9 @@ def escape_redshift_identifier(v: str) -> str:
 def escape_bigquery_identifier(v: str) -> str:
     # https://cloud.google.com/bigquery/docs/reference/standard-sql/lexical
     return "`" + v.replace("\\", "\\\\").replace("`","\\`") + "`"
+
+
+def escape_snowflake_identifier(v: str) -> str:
+    # Snowcase uppercase all identifiers unless quoted. Match this here so queries on information schema work without issue
+    # See also https://docs.snowflake.com/en/sql-reference/identifiers-syntax#double-quoted-identifiers
+    return v.upper()
```

Comment:
we asked our snowflake buddies what they use typically. let's see what they say

Reply:
Forgot to take into account escaping sql keywords also, e.g. : `CREATE TABLE xxx (FROM VARCHAR)`  
Thinking now we should both quote and upcase.

Reply:
it seems that people mostly ignore it: use the original casing without quotes. this of course translates to upper case internally. that for example makes dbt packages to look identical for different systems. I'm still waiting for more input. sql keywords? aren't they case insensitive as well?

Reply:
I mean with keywords if you have a column named `from` or `update` or something. Those need to be quoted.

So I think if we uppercase and quote in the escape function: `some_column` -> `"SOME_COLUMN"` makes things work pretty seamlessly, least resistance to how snowflake works.

Then queries with unquoted cols like `SELECT some_column FROM ...` work since snowflake translates it to uppercase implicitly.